RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,233 @@
+* [RFC-0340: Validator Node Consensus](RFC-0340_VNConsensusOverview.md)
+* [RFC-0374: Asset Creation and management](RFC-0374_TCAssetManagement.md)
+
+## Description
+
+### Motivation
+
+The reasons for issuing assets on Tari under a templating system, rather than a scripting language (whether Turing
+complete or not) are manifold:
+
+* A scripting language, irrespective of how simple it is, limits the target market for asset issuers to developers, or
+  people who pay developers.
+* The market doesn’t want general smart contracts. This is evidenced by the fact that the vast majority of Ethereum
+  transactions go through ERC-20 or ERC-721 contracts, which are literally contract templates.
+* The attack surface for smart contracts is reduced considerably; to the node software itself.
+* Bugs can be fixed for all contracts simultaneously by using a template versioning system. Existing assets can opt-in
+  to fixes by migrating assets to a new version of the contract.
+* Contracts will have better QA since more eyes are looking at fewer contract code sets.
+* Transmission, storage and processing of contracts will be more efficient as one only has to deal with the parameters
+  and not the logic of the contract. Furthermore, the cost for users is usually lower, since there's no need to add
+  friction / extra costs to contract execution (e.g. ethereum gas) to work around the halting problem.
+
+### Implementation
+
+Assets are created on the Tari network by issuing a `create_asset` instruction from a wallet or client and broadcasting
+it to the Tari Digital Assets Network (DAN).
+
+The instruction MUST contain the following fields:
+(PKH = Public Key Hash)
+
+| Name                               | Type          | Description                                                                                                                     |
+|:-----------------------------------|:--------------|:--------------------------------------------------------------------------------------------------------------------------------|
+| **Asset description**              |               |                                                                                                                                 |
+| issuer                             | PKH           | The public key of the creator of the asset. See [issuer](#issuer)                                                               |
+| name                               | `string[64]`  | The name or identifier for the asset. See [Name and Description](#name-and-description)                                         |
+| description                        | `string[216]` | A short description of the asset - with name, fits in a tweet. See [Name and Description](#name-and-description)                |
+| tldi                               | `string[16]`  | The Top Level Digital Issuer (TLDI) for the asset. The TLDI must be owned by the asset issuer, or "" for an unregistered asset  |
```

Comment:
`TLDI` was changed to `RAID_ID`
string[16] will be too small, looks more like 44 hex characters, thus string[64]
Registered Asset Issuer Domain (RAID)
RAID_ID (RAID Identification)

Reply:
An empty string does not hash well
I am currently proposing `RAID_ID = Base58Check(Hash256("No FQDN"))` and 
`RAID_ID = Base58Check(Hash256(PubKey || FQDN))` in
RFC-0301/NamespaceRegistration, PR to follow soon

Reply:
64 characters for a purely informational ID field is too much; I suggest we look at ways to represent the RAID id in 16 characters, either by clipping or using a different hash function.

Reply:
Le us discuss this, current proposal has RAID_ID at 34 characters.

Reply:
Discussion in #150